Documentation/contributing/docs/docsframework.rst

```diff
@@ -0,0 +1,382 @@
+``Documentation/conf.py``.
+
+.. _spell-checker module: https://github.com/sphinx-contrib/spelling
+.. _builder: https://www.sphinx-doc.org/en/master/usage/builders
+
+:spelling:ignore:`rstcheck`
+---------------------------
+
+We use `rstcheck`_ to validate the rST formatting. We have a list of warnings
+to ignore (in part because the linter has bugs). The way we call the tool, and
+this list of exceptions, are configured in ``Documentation/check-build.sh``.
+
+.. _rstcheck: https://rstcheck.readthedocs.io
+
+Link checker
+------------
+
+We have a link checker under ``Documentation/check-links.sh`` for the
+documentation. However, due to some unsolved issues, we do not run it in CI.
+See :gh-issue:`27116` for details.
+
+Web server for local preview
+----------------------------
+
+Launch a web server to preview the generated documentation locally with ``make
+render-docs``.
+
+See more on this topic on the page about :ref:`testing documentation
+<testing-documentation>` page.
+
+Custom Sphinx roles
+-------------------
+
+We use several custom roles in the documentation:
+
+- ``git-tree``
+- ``github-project``
+- ``github-backport``
+- ``gh-issue``
+- ``prev-docs``
+
+We use these roles to insert links based on specific URL templates, via the
+`extlinks`_ extension. They are all configured in ``Documentation/conf.py``.
+They should be used wherever relevant, to ensure that the way we link to
+different resources remains consistent.
+
+.. _extlinks: https://www.sphinx-doc.org/en/master/usage/extensions/extlinks.html
+
+Custom Sphinx directives
+------------------------
+
+We do not implement custom directives as of this writing.
+
+Custom extensions
+-----------------
+
+We use custom extensions for Sphinx, implemented under ``Documentation/_exts``.
+
+- One defines the custom filters for the spell checker.
+- One patches Sphinx's HTML translator to open all external links in new tabs.
+
+Google Analytics
+----------------
+
+The documentation uses Google Analytics to collect metrics. This is configured
+in ``Documentation/conf.py``.
+
+Customization
+-------------
+
+Additional elements of customization for Cilium's documentation and defined in
+the main repository include:
+
+- Some custom CSS; see also class ``wrapped-table`` in the related CSS file
+  ``Documentation/_static/wrapped-table.css``
+
+- A "Copy" button, including a button to copy only commands from console-code
+  blocks, implemented in ``Documentation/_static/copybutton.js`` and
+  ``Documentation/_static/copybutton.css``
+
+- Custom header and footer definitions, for example to make link to Slack
+  target available on all pages
+
+- Warning banner on older branches, telling to check out the latest version
+  (these may be handled directly in the ReadTheDocs configuration in the
+  future, see also :gh-issue:`29969`)
+
+:spelling:ignore:`Algolia` search engine
+----------------------------------------
+
+- *To be documented.*
```

Comment:
Is there something custom we do here or is Algolia just crawling the versions we configure in https://github.com/cilium/docsearch-scraper-webhook ?

Reply:
I have no idea whatsoever how we work with Algolia. This is the reason why this section is empty :innocent:.

Reply:
Me neither. It's a mystery!

Reply:
I don't know, I don't quite see the doc introduced in this PR as something that describes the “docs contribution workflows”. In my mind it's more something that describes the current resources that we have and use for building the documentation. The typical target audience would not be someone trying to do a “regular” contributions to the docs (by this, I mean a contribution to docs contents, for example), but more someone who looks into the internals of “what do these components do and how does it all connect together to render the website”. I believe the doc does a decent job at describing the individual components, even if “connecting them together” could likely be improved in future iterations.

As such, I see Algolia as one of these components that we use to build Cilium's docs. It's true that it usually has no impact on contributions, which also explains why I've never been much exposed to it, but to me it still made sense to have it listed in this document, with the hope that someone with more knowledge (maybe myself in the future?) might pick it up and complete it. In the meantime, I could link to https://github.com/cilium/docsearch-scraper-webhook in that subsection, if you think it makes sense. Or I can also remove the section if you think strongly about it.

Reply:
I linked to the scraper webhook and updated my branch. All other feedback have been addressed, I'll send the PR to the merge queue. Please do let me know if you feel like this deserves a clarification/deletion/... in a follow-up PR.

Reply:
That makes sense to me. From that perspective, having _something_ is better than having nothing at all. The changes you made LGTM. :+1: